homeassistant/components/tado/climate.py

```diff
@@ -435,11 +474,24 @@ def set_swing_mode(self, swing_mode):
     def _async_update_zone_data(self):
         """Load tado data into zone."""
         self._tado_zone_data = self._tado.data["zone"][self.zone_id]
+        self._update_offset()
         self._current_tado_fan_speed = self._tado_zone_data.current_fan_speed
         self._current_tado_hvac_mode = self._tado_zone_data.current_hvac_mode
         self._current_tado_hvac_action = self._tado_zone_data.current_hvac_action
         self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
 
+    def _update_offset(self):
```

Comment:
Looks like we don't need this method. We only call it from a single place. We can inline the offset update inside `_async_update_zone_data`.

Reply:
Yeah fair point, can 't remember why I thought a dedicated method was needed, updated